extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -548,8 +548,10 @@ export class AmpConsent extends AMP.BaseElement {
       const expireCache = response['expireCache'];
       if (expireCache) {
         this.consentStateManager_.setDirtyBit();
-        // Decision from promptUI takes precedence over consent decision from response
-      } else if (
+      }
+
+      // Decision from promptUI takes precedence over consent decision from response
+      if (
```

Comment:
will this override `setDirtyBit` ?

Reply:
No, it will update the values with the dirty bit, which will clear the values.

Reply:
I'm a bit confused by this code:
https://github.com/ampproject/amphtml/blob/63fa1d05386d20ff6135a42000cf9d3036b253cd/extensions/amp-consent/0.1/consent-state-manager.js#L296-L312

Just to be safe, shall we just move `setDirtyBit` to after the `updateCacheIfNotNull_` call

Reply:
`setDirtyBit` sets a local value that will be used when writing value to localStorage. So it's fine to set the dirtyBit earlier. 

However there's a performance concern here. Setting dirtyBit requires a localStorage read that could be combined with the write right afterwards. I'm not sure what the save can be, but worth improving the code later.

Reply:
Good point, definitely worth looking into later.